src/frontend/Users.tsx

```diff
@@ -32,36 +43,60 @@ export const Users: React.FunctionComponent<RouteComponentProps> = () => {
   }) => {
     try {
       await putRequest(`/api/users/${uuid}/role`, { role }, { useJwt: true });
+
+      // Update the state
+      const i = users.findIndex((u) => u.uuid === uuid);
+      const usersUpdated = [...users];
+      usersUpdated[i].role = role;
+      setUsers(usersUpdated);
+
       addNotification({ type: 'success', message: `Updated ${email} role to ${role}` });
     } catch (err) {
       console.log(err);
       addNotification({ type: 'error', message: `Failed to update ${email} role to ${role}` });
     }
   };
 
-  useEffect(() => {
-    const fetchUsers = async () => {
+  const fetchUsers = useCallback(
+    async (callback: (allUsers: Array<ApiUserData>) => void) => {
       try {
         const result = await getRequest<{ data: Array<ApiUserData> }>(`/api/users`, {
           useJwt: true,
         });
-        setUsers(result.data.data);
+        const data = result.data.data;
+        setUsers(data);
```

Comment:
Is setUsers now called twice, because it seems to also be called in the useEffect on line 81? Anyways not a big problem.